packages/addons/service/locale/package.mk

```diff
@@ -16,13 +16,13 @@ PKG_ADDON_TYPE="xbmc.service"
 
 addon() {
   mkdir -p "$ADDON_BUILD/$PKG_ADDON_ID/bin"
-  cp -PR "$(get_build_dir glibc)/.$TARGET_NAME/locale/localedef" \
+  cp -PR "$(get_install_dir glibc)/.noinstall/localedef" \
          "$ADDON_BUILD/$PKG_ADDON_ID/bin"
 
-  mkdir -p "$ADDON_BUILD/$PKG_ADDON_ID/i18n"
-  cp -PR "$(get_build_dir glibc)/localedata/charmaps" \
-         "$(get_build_dir glibc)/localedata/locales" \
-         "$ADDON_BUILD/$PKG_ADDON_ID/i18n"
+  cp -PR "$(get_install_dir glibc)/usr/share/i18n" \
+         "$(get_install_dir glibc)/.noinstall/charmaps" \
+         "$(get_install_dir glibc)/.noinstall/locales" \
+         "$ADDON_BUILD/$PKG_ADDON_ID"
```

Comment:
Wow... the way `glibc` deals with `charmaps` and `locales` is a bit of a mess. As such I'm still not entirely sure this is correct!

Previously we were creating the `i18n` directory in the root of the addon, and into the `i18n` directory we copied the `charmaps` and `locales` as sub-directories.

However the code above will copy the `i18n` directory (which will include a `locales` sub-dir, the content of which will depend on `$GLIBC_LOCALES` - it will either be the GLIBC or POSIX locales, and a `UTF-8` only `charmaps` sub-dir for `Generic` only - won't exist for non-Generic) into the root. And then it will also copy the full GLIBC `charmaps` and `locales` into the root of the addon. That's all 3 directories in the root of the addon (with up to 2 sub-dirs in `i18n`), which is a different structure to previously.

I think we only want to create an empty i18n folder, and then into `i18n` copy only the `charmaps` and `locales` directories from `.noinstall` - which will be the full GLIBC `charmaps` and `locales`. We do not want to copy `/usr/share/i18n` as that potentially brings with it the `utf-8` only `charmaps` and/or POSIX-only `locales`.

It's not entirely clear if we should be using the POSIX-only locales in the add-on when `GLIBC_LOCALES` is `no` - my guess is that we probably should, in which case the content of `.noinstall/locales` would need to reflect that.

Reply:
The directory mess wasn't intended, fixed, thanks

Reply:
Sorry I didn't mean that to sound like I was blaming you - I think this is just a historical accumulation of cruft and this PR tipped it over the edge! :)

The latest `locale` and `glibc` changes look good to me, thanks!

Reply:
Don't worry about it, no blaming was involved! If anything I'm annoyed at myself for getting sloppy there ;)